README.md

```diff
@@ -167,10 +172,51 @@ usage is clear you can use `U`, `V` etc. for any following generic.
 If the usage is not obvious, you should use a more descriptive name. The same naming rules as for
 classes will apply then.
 
-#### Prefixes
+```ts
+// bad, prefix generics with T
+class Response<TResponseData> {
+  public data: TResponseData;
+}
 
-We are not using prefixes for any name. Interfaces should follow the exact naming rules as classes,
-and should not use the `I` or any other pre- or postfix. 
+// good, use common abbreviations like T(ype), K(ey), V(alue), P(roperty) etc.
+type Partial<T> = { [P in keyof T]?: T[P]; };
+
+// better, add more semantic context by extending the type
+type ResponseData = Record<string, unknown>;
+
+class Response<T extends ResponseData> {
+  public data: T;
+}
+```
+
+#### Interfaces and type aliases
+Depending on the use case there is a choice to implement an
+[interface or type alias](https://github.com/typescript-cheatsheets/react#useful-table-for-types-vs-interfaces),
+but be aware using types impacts [compilation performance](https://ncjamieson.com/prefer-interfaces/).
+
+
+```ts
+// bad, prefix interfaces with I
```

Comment:
nitpick, but the language is a bit confusing to me here, when I read this comment in isolation, I think _the below is bad, I should prefix interfaces with I_. Might be just the way I read it, but maybe we can just keep it as `// bad` and `// good` ?

For example, that's how eslint does it: https://github.com/eslint/eslint/blob/master/docs/rules/accessor-pairs.md

Reply:
agreed

```suggestion
// bad, you should not prefix interfaces with I
```